### ISSUE TITLE
JupyterLab puts confusing prompt in input field for `input` and `getpass`

### ISSUE THREAD

**Issue by paulmromer** [jupyterlab/jupyterlab#15977]
## Description

JupyterLab displays inappropriate text in the input field for `input` and `getpass`. 

The text shows only when the cursor is in the input field. If you click out, the field is empty, as it should be. 

The specific text that shows up is different for different versions of JupyterLab: 

JupyterLab 4.1.4:
![initial_state_or_click_in_JL-4 1 4](https://github.com/jupyterlab/jupyterlab/assets/33130966/0a4204c9-1c27-4a8a-b1fb-b62a87d8d1d1)

JupyterLab 3.6.2 
![input_initial_state_or_click_in_JL-3 6 2](https://github.com/jupyterlab/jupyterlab/assets/33130966/ab4726b8-bfda-4471-b6e4-469cfdf2c8b0)

In both cases, the cursor blinks. These two images differ because one caught the cursor when it was on, the other caught it when it was off. 

## Reproduce

1. Put this command into an input cell  `input()` and run it. Or include this import `from getpass import getpass` and put `getpass()` into an input cell and run it. 

FYI, I have reproduced this only on macOS. As noted above, the problem shows up differently on different versions of JupyterLab.

I discovered this first with the `getpass` command. The confusing text in the input field is more problematic in that use case. One should be worried by a request for a password that looks strange. Then I replicated the behavior with the input command. 

## Expected behavior

I do not recall trying this in earlier versions of JupyterLab. But based on the behavior in a *.py file, I would expect the input field to be empty. Also, with `getpass`, on the basis of how it works in a *.py file, I would expect to see the default prompt "Password:" before the field in the default case that specifies no prompt but in fact, in the default cause there is no prompt: 

![password_default_prompt_cursor_in_field](https://github.com/jupyterlab/jupyterlab/assets/33130966/884f18c4-0723-4feb-a98c-371e35d5dd45)

![password_prompt_cursor_in_field](https://github.com/jupyterlab/jupyterlab/assets/33130966/89436692-2f7b-4598-ac09-1789f48176c8)

## Context

- Operating System and version: macOS 14.4
- Browser and version:  Chrome 122.0.6261.112 , Safari 17.4, Firefox 123.0.1 (all with JL 4.1.4)
- JupyterLab version:  4.1.4 and 3.6.2 (with Firefox 123.0.1)

## File with the text that shows up in the Field

jlab_core.4f5a04974adab2b69643.js

## Browser Output

![from_browser](https://github.com/jupyterlab/jupyterlab/assets/33130966/3b8442c0-f961-4339-9a3c-bc9015e7495d)

In the browser console, editing the HTML to set placeholder = "" fixes the issue temporarily.

**Comment by krassowski** [comment]
This was implemented in https://github.com/jupyterlab/jupyterlab/pull/12502 that was backported to 3.6.x. The placeholder was then modified for 4.0 in https://github.com/jupyterlab/jupyterlab/pull/13937.

There is an existing issue asking for [an option to disable input placeholder text](https://github.com/jupyterlab/jupyterlab/issues/15138).

I agree that showing this placholder in password fields is awkward. On the other hand, the placholder is needed to expose the feature from https://github.com/jupyterlab/jupyterlab/issues/7006 which was frequently requested.

Please note that placeholder and prompt serve related but different purposes. You could have a `e-mail:` prompt and `your@mail.com` prompt which would render like:

```
e-mail: [ your@mail.com        ]
```

**Comment by krassowski** [comment]
Setting placeholder conditionally on whether `options.password` is set should be an easy self-contained change around

https://github.com/jupyterlab/jupyterlab/blob/910a16229bd75553baaa5c2a545e3fa798234c75/packages/outputarea/src/widget.ts#L1094-L1098

**Comment by paulmromer** [comment]
Sorry I missed the other issue related to input -- I only looked for issues related to `getpass`. 

If I understand your second suggestion correctly, a fix for `getpass` only would be great as far as I'm concerned. I have no reason to believe that there should be a change that applies to the `input` command. I'd have to look a lot more carefully before weighing in on that. Sorry if I seemed to suggest otherwise. 

Also, for `getpass`, I don't think it matters much at all whether the default prompt "Password:" shows up when no prompt is supplied so the user needs to add it explicitly.

**Comment by krassowski** [comment]
Would you be interested in opening a pull request? It would need to wrap a few bits of logic in `if (!options.password)`.

**Comment by paulmromer** [comment]
Yes. Thx. 

I inferred from your comments that it might be appropriate for me to submit a pull request. 

I hesitated because I wanted to some homework. I have literally never even looked at code written in Typescript. I will try to look at this over the weekend. I'm also looking for someone who can check anything I do first before I submit it. 

PS: Thanks for both this suggestion about what the if statement would look like. Also for the clarification about the distinction between prompt and placeholder.
